fuzzers/main.py

```diff
@@ -0,0 +1,819 @@
+        branch = self.branch['name']
+        logs_url = f'https://storage.cloud.google.com/fuzzer/logs/{self.log_relpath}'
+
+        # Check the artifact was not reported yet for this branch
+        if artifact in REPORTED_ARTIFACTS[branch]:
+            return
+        already_reported_for = [
+            other_branch for (other_branch,
+                              reported_artifacts) in REPORTED_ARTIFACTS.items()
+            if artifact in reported_artifacts
+        ]
+        REPORTED_ARTIFACTS[branch].append(artifact)
+
+        if already_reported_for:
+            already_reported_msg = f'(already reported for branches {already_reported_for})'
+        else:
+            already_reported_msg = ''
+
+        # Send the information in two mesages, a short one guaranteed to succeed, then a long one
+        # with the log lines that could go over the message size limit
+        client = zulip.Client(config_file=ZULIPRC)
+        gcs_artifact = shlex.quote(
+            f'gs://{bucket.name}/{self.corpus_vers}/{self.target["crate"]}/'
+            f'{self.target["runner"]}/artifacts/{artifact}')
+        artifact_dest = shlex.quote(
+            f'{self.target["crate"]}/artifacts/{self.target["runner"]}/{artifact}'
+        )
+        quoted_crate = shlex.quote(self.target['crate'])
+        quoted_runner = shlex.quote(self.target['runner'])
+        quoted_artifact = shlex.quote(
+            f'artifacts/{self.target["runner"]}/{artifact}')
+        client.send_message({
+            'type':
+                'stream',
+            'to':
+                'nearinc/fuzzer/private',
+            'topic':
+                f'{branch}: artifact {artifact}',
+            'content': (
+                f'# Fuzzer found new crash for branch *{branch}* {already_reported_msg}\n'
+                f'Full logs are available at {logs_url}.\n'
+                f'\n'
+                f'You can download the artifact by using the following command (all commands '
+                f'are to be run from the root of `nearcore`):\n'
+                f'```\n'
+                f'gsutil cp {gcs_artifact} {artifact_dest}\n'
+                f'```\n'
+                f'\n'
+                f'Then, you can reproduce by running the following command:\n'
+                f'```\n'
+                f'cd {quoted_crate}\n'
+                f'RUSTC_BOOTSTRAP=1 cargo fuzz run {quoted_runner} {quoted_artifact}\n'
+                f'```\n'
+                f'\n'
+                f'Or minimize by running the following command:\n'
+                f'```\n'
+                f'cd {quoted_crate}\n'
+                f'RUSTC_BOOTSTRAP=1 cargo fuzz tmin {quoted_runner} {quoted_artifact}\n'
+                f'```\n'
+                f'\n'
+                f'Please edit the topic name to add more meaningful information once investigated. '
+                f'Keeping the artifact hash in it can help if the same artifact gets detected '
+                f'as crashing another branch.\n'),
```

Comment:
I’d use f''' ... ''' here:

```python
            'content': f'''\
# Fuzzer found new crash for branch *{branch}* {already_reported_msg}
Full logs are available at {logs_url}.

You can download the artifact by using the following command (all
command are to be run from the root of `nearcore`):

```
gsutil cp {gcs_artifact} {artifact_dest}
```

Then, you can reproduce by running the following command:

```
cd {quoted_crate}
RUSTC_BOOTSTRAP=1 cargo fuzz run {quoted_runner} {quoted_artifact}
```

Or minimize by running the following command:

```
cd {quoted_crate}
RUSTC_BOOTSTRAP=1 cargo fuzz tmin {quoted_runner} {quoted_artifact}
```

Please edit the topic name to add more meaningful information once
investigated.  Keeping the artifact hash in it can help if the same
artifact gets detected as crashing another branch.
''',
```